From 36b7e7a70b56cffbf481e79b177e31caa0b82587 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Thu, 28 Mar 2013 17:01:19 +0800
Subject: [PATCH 1/4] ENGR00256175 Android reboot if setup multi-user with
 space running out

Only send MESSAGE_SETUP_USER MSG if user created successfully with valid
userinfo.

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 src/com/android/settings/users/UserSettings.java | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/com/android/settings/users/UserSettings.java b/src/com/android/settings/users/UserSettings.java
index bbae37d..89e0406 100644
--- a/src/com/android/settings/users/UserSettings.java
+++ b/src/com/android/settings/users/UserSettings.java
@@ -584,11 +584,15 @@ public class UserSettings extends RestrictedSettingsFragment
                         mAddingUser = false;
                         if (userType == USER_TYPE_USER) {
                             mHandler.sendEmptyMessage(MESSAGE_UPDATE_LIST);
-                            mHandler.sendMessage(mHandler.obtainMessage(
+                            if (user != null) {
+                                mHandler.sendMessage(mHandler.obtainMessage(
                                     MESSAGE_SETUP_USER, user.id, user.serialNumber));
+                            }
                         } else {
-                            mHandler.sendMessage(mHandler.obtainMessage(
+                            if (user != null) {
+                                mHandler.sendMessage(mHandler.obtainMessage(
                                     MESSAGE_CONFIG_USER, user.id, user.serialNumber));
+                            }
                         }
                     }
                 }
-- 
1.8.0

